packages/synthetic-chain/README.md

```diff
@@ -0,0 +1,29 @@
+# Synthetic chain tools
+
+Utilities to build a synthetic chain and test running proposals atop it. The chain approximates agoric-3 (Mainnet) using the state from https://github.com/Agoric/agoric-3-proposals (It could trivially support other Agoric chains, if we scale horizontally.)
+
+## Design
+
+Builds atop the `main` image of https://ghcr.io/agoric/agoric-3-proposals to execute proposals on top of the agoric-3 chain and also test them.
+
+It also adopts the multi-stage Docker build flow from the a3p repo. See https://github.com/Agoric/agoric-3-proposals?tab=readme-ov-file#design
+
+One deficiency in the current design is that share code in `upgrade-test-scripts` is not versioned or packaged. Any change to it will invalidate the base image. And each layer assumes it has the latest. And now with this repo, the local copy has to be kept in sync with the base image and the a3p repo.
```

Comment:
> mean that the shared scripts have to remain backwards compatible, but I don't think that's a huge constraint

Yet is is a constraint.

> Consider it the "API" offered by the a3p image

That's a reasonable model. But I think a versioned package defining the API is better.

Right now when we change these scripts, all the layers have to be rebuilt. That's been a pain while developing, so I wanted these versioned too. I failed to update this part of the README: this PR now versions the scripts by including them in the package. The whole a3p repo will use this package as its tooling: https://github.com/Agoric/agoric-3-proposals/pull/46 This package defines the API and the image is a built dependency.

Reply:
Echoing our conversation, it's better to move the package out now:
- in agoric-sdk it could only publish with each SDK release (save for `dev` builds)
- in agoric-sdk, it could complicate cherry-picking and other tooling
- it's not a big left to get agoric-3-proposals to publish

I've marked this PR draft while I move the package over to a3p. After it's published in NPM I'll make this PR have just the migration of `upgrades-tests` to `a3p-integration`, using the published package.